src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/ValueLayout.java

```diff
@@ -540,66 +549,72 @@ public OfAddress withOrder(ByteOrder order) {
-     * bit-alignment set to 8, and byte order set to {@link ByteOrder#nativeOrder()}.
+     * bit-alignment set to 32, and byte order set to {@link ByteOrder#nativeOrder()}.
      * Equivalent to the following code:
      * <blockquote><pre>{@code
-    MemoryLayout.valueLayout(int.class, ByteOrder.nativeOrder()).withBitAlignment(8);
+    MemoryLayout.valueLayout(int.class, ByteOrder.nativeOrder()).withBitAlignment(32);
      * }</pre></blockquote>
      */
-    public static final OfInt JAVA_INT = new OfInt(ByteOrder.nativeOrder()).withBitAlignment(8);
+    public static final OfInt JAVA_INT = new OfInt(ByteOrder.nativeOrder())
+            .withBitAlignment(32);
 
     /**
      * A value layout constant whose size is the same as that of a Java {@code long},
-     * bit-alignment set to 8, and byte order set to {@link ByteOrder#nativeOrder()}.
+     * bit-alignment set to 64, and byte order set to {@link ByteOrder#nativeOrder()}.
      * Equivalent to the following code:
      * <blockquote><pre>{@code
-    MemoryLayout.valueLayout(long.class, ByteOrder.nativeOrder()).withBitAlignment(8);
+    MemoryLayout.valueLayout(long.class, ByteOrder.nativeOrder()).withBitAlignment(64);
      * }</pre></blockquote>
      */
     public static final OfLong JAVA_LONG = new OfLong(ByteOrder.nativeOrder())
-            .withBitAlignment(8);
+            .withBitAlignment(ADDRESS_SIZE_BITS);
 
     /**
      * A value layout constant whose size is the same as that of a Java {@code float},
-     * bit-alignment set to 8, and byte order set to {@link ByteOrder#nativeOrder()}.
+     * bit-alignment set to 32, and byte order set to {@link ByteOrder#nativeOrder()}.
      * Equivalent to the following code:
      * <blockquote><pre>{@code
-    MemoryLayout.valueLayout(float.class, ByteOrder.nativeOrder()).withBitAlignment(8);
+    MemoryLayout.valueLayout(float.class, ByteOrder.nativeOrder()).withBitAlignment(32);
      * }</pre></blockquote>
      */
-    public static final OfFloat JAVA_FLOAT = new OfFloat(ByteOrder.nativeOrder()).withBitAlignment(8);
+    public static final OfFloat JAVA_FLOAT = new OfFloat(ByteOrder.nativeOrder())
+            .withBitAlignment(32);
 
     /**
      * A value layout constant whose size is the same as that of a Java {@code double},
-     * bit-alignment set to 8, and byte order set to {@link ByteOrder#nativeOrder()}.
+     * bit-alignment set to 64, and byte order set to {@link ByteOrder#nativeOrder()}.
      * Equivalent to the following code:
      * <blockquote><pre>{@code
-    MemoryLayout.valueLayout(double.class, ByteOrder.nativeOrder()).withBitAlignment(8);
+    MemoryLayout.valueLayout(double.class, ByteOrder.nativeOrder()).withBitAlignment(64);
      * }</pre></blockquote>
      */
-    public static final OfDouble JAVA_DOUBLE = new OfDouble(ByteOrder.nativeOrder()).withBitAlignment(8);
+    public static final OfDouble JAVA_DOUBLE = new OfDouble(ByteOrder.nativeOrder())
+            .withBitAlignment(ADDRESS_SIZE_BITS);
```

Comment:
The code here is still using `ADDRESS_SIZE_BITS` (same for JAVA_LONG), instead of 64.